tests/pos/i18234.scala

```diff
@@ -0,0 +1,38 @@
+//> using options -Xfatal-warnings -Wdollar-check
+
+val goodVal = 1
```

Comment:
Please add pos tests for unnamed givens (e.g. `given String = "foo"`), lazy vals, implicit params coming from `:` syntax (e.g. `def foo[T: Showable]`), partial function definition

Reply:
Also some anonymous functions and a companion object. We have to make sure that no synthetic symbols are catched by linting. A good thing is that we can may be able to access the sources to confirm if the symbol is present, but we will do it only if necessary (however, I suspect it may be).

Reply:
I think I have covered all the cases you mentioned now. It actually helped catch a case for anonymous functions. Which I added a fix for.

Reply:
actually, after adding this phase to the same list as `CheckUnused.PostTyper` made it be triggered on compilation of the project, and in turn revealed a lot of false positives. I'll try and look into them this week